warehouse/query-core/src/test/java/datawave/query/function/HitsAreAlwaysIncludedTest.java

```diff
@@ -381,37 +458,107 @@ public void testHitWithExceededOrThreshold() throws Exception {
     }
     
     @Test
-    public void testHitWithFunction() throws Exception {
+    public void testHitsOnly() throws Exception {
         Map<String,String> extraParameters = new HashMap<>();
         extraParameters.put("include.grouping.context", "false");
         extraParameters.put("hit.list", "true");
-        extraParameters.put("limit.fields", "FOO_1_BAR=3,FOO_1=2,FOO_3=2,FOO_3_BAR=4,FOO_4=3,FOO_1_BAR_1=4");
+        extraParameters.put("limit.fields", "FOO_1_BAR=-1,FOO_1=-1,FOO_3=-1,FOO_3_BAR=-1,FOO_4=-1,FOO_1_BAR_1=-1");
         logic.setMaxOrExpansionThreshold(1);
         ivaratorConfig();
         
         String queryString = "FOO_3_BAR == 'defg<cat>' || FOO_3_BAR == 'abcd<cat>'";
         
         // there is no grouping context so i can expect only the original term, not the related ones (in the same group)
+        //@formatter:off
         Set<String> goodResults = Sets.newHashSet("FOO_3_BAR:defg<cat>", "FOO_3_BAR:abcd<cat>");
+        //@formatter:on
         Set<String> expectedHits = Sets.newHashSet("FOO_3_BAR:defg<cat>", "FOO_3_BAR:abcd<cat>");
         
         runTestQuery(queryString, format.parse("20091231"), format.parse("20150101"), extraParameters, expectedHits, goodResults);
     }
     
+    @Test
+    public void testGroupedHitsOnly() throws Exception {
+        Map<String,String> extraParameters = new HashMap<>();
+        extraParameters.put("include.grouping.context", "true");
+        extraParameters.put("hit.list", "true");
+        extraParameters.put("limit.fields", "FOO_1_BAR=-1,FOO_1=-1,FOO_3=-1,FOO_3_BAR=-1,FOO_4=-1,FOO_1_BAR_1=-1");
+        logic.setMaxOrExpansionThreshold(1);
+        ivaratorConfig();
+        
+        String queryString = "FOO_3_BAR == 'defg<cat>' || FOO_3_BAR == 'abcd<cat>'";
+        //@formatter:off
+        Set<String> goodResults = Sets.newHashSet("FOO_1_BAR.FOO.3:good<cat>", "FOO_3_BAR.FOO.3:defg<cat>", "FOO_3.FOO.3.3:defg", "FOO_4.FOO.4.3:yes",
+                "FOO_1.FOO.1.3:good",
+                "FOO_1.FOO.1.0:yawn",
+                "FOO_4.FOO.4.0:purr",
+                "FOO_3.FOO.3.0:abcd",
+                "FOO_3_BAR.FOO.0:abcd<cat>",
+                "FOO_1_BAR.FOO.0:yawn<cat>",
+                "FOO_1_BAR_1.FOO.0:2021-03-24T16:00:00.000Z");
+        //@formatter:on
+        Set<String> expectedHits = Sets.newHashSet("FOO_3_BAR.FOO.3:defg<cat>", "FOO_3_BAR.FOO.0:abcd<cat>");
+        
+        runTestQuery(queryString, format.parse("20091231"), format.parse("20150101"), extraParameters, expectedHits, goodResults);
+    }
+    
+    @Test
+    public void testGroupedHitsWithMatchingField() throws Exception {
+        Map<String,String> extraParameters = new HashMap<>();
+        extraParameters.put("include.grouping.context", "true");
+        extraParameters.put("hit.list", "true");
+        extraParameters.put("limit.fields", "FOO_1_BAR=-1,FOO_1=-1,FOO_3=-1,FOO_3_BAR=-1,FOO_4=-1,FOO_1_BAR_1=-1");
+        extraParameters.put("matching.field.sets", "FOO_4");
```

Comment:
A test with multiple matching field sets would be good.

Reply:
Very true.  I will work that.  Thanks.

Reply:
thank you!